src/org/gk/slicing/RevisionDetector.java

```diff
@@ -0,0 +1,442 @@
+			throws InvalidAttributeException, Exception {
+		if (pathway == null)
+			return false;
+
+		// Check if an immediate child Pathway is added or removed
+		if (isChildPathwayAddedOrRemoved(pathway, targetDBA))
+			return true;
+
+		// Finally check for changes in summation text.
+		if (isSummationRevised(pathway, targetDBA))
+			return true;
+
+		// Recursively iterate over events in pathway.
+		Collection<GKInstance> events = pathway.getAttributeValuesList(ReactomeJavaConstants.hasEvent);
+
+		if (events != null && events.size() > 0) {
+			for (GKInstance event : events) {
+				// Pathway
+				if (event.getSchemClass().isa(ReactomeJavaConstants.Pathway))
+					isPathwayRevised(event, targetDBA);
+
+				// RLE
+				else if (event.getSchemClass().isa(ReactomeJavaConstants.ReactionlikeEvent)
+					 && isRLERevised(event, targetDBA))
+					return true;
+			}
+		}
+
+		return false;
+	}
+
+	/**
+	 * Check if a child pathway is added or removed for a given parent.
+	 *
+	 * @param parent
+	 * @return boolean (true if a child pathway is added or removed, false otherwise).
+	 * @throws InvalidAttributeException
+	 * @throws SQLException
+	 * @throws Exception
+	 */
+	private boolean isChildPathwayAddedOrRemoved(GKInstance parent, MySQLAdaptor targetDBA)
+			throws InvalidAttributeException, SQLException, Exception {
+
+		Collection<GKInstance> childPathways = getChildPathways(parent);
+		Collection<GKInstance> compareChildPathways = getChildPathways(getInstance(targetDBA, parent));
+
+		if (childPathways.isEmpty() && compareChildPathways.isEmpty())
+			return false;
+
+		// If the number of child pathways is different between the two parents,
+		// then an addition or deletion has taken place.
+		if (childPathways.size() != compareChildPathways.size())
+			return true;
+
+		// List of child pathway IDs.
+		Collection<Long> childPathwayDBIDs = childPathways.stream()
+				.map(attrValue -> attrValue.getDBID())
+				.collect(Collectors.toList());
+
+		// Check if there is a child pathway not associated with both parents.
+		Optional<Long> commonDBID = compareChildPathways.stream()
+				// Get database ID.
+				.map(attrValue -> attrValue.getDBID())
+				// Set match criteria (identical database ID's).
+				.filter(comparePathwayDBID -> childPathwayDBIDs.contains(comparePathwayDBID))
+				// Find match (or null if no match).
+				.findAny();
+
+		if (!commonDBID.isPresent())
+			return true;
+
+		return false;
+	}
+
+	/**
+	 * Return a list of child pathways for a given parent.
+	 *
+	 * @param parent
+	 * @return List
+	 * @throws InvalidAttributeException
+	 * @throws Exception
+	 */
+	private Collection<GKInstance> getChildPathways(GKInstance parent) throws InvalidAttributeException, Exception {
+		Collection<GKInstance> events = parent.getAttributeValuesList(ReactomeJavaConstants.hasEvent);
+		return events.stream()
+				.filter(attr -> attr.getSchemClass().isa(ReactomeJavaConstants.Pathway))
```

Comment:
ReactionlikeEvent should be considered too. There is no need to have this method. Just call pathway.getAttributeValuesList() should be enough.

Reply:
Fixed in [5224afe](https://github.com/reactome/CuratorTool/pull/20/commits/5224afe9bbc88b7d82e68e63a4632c6fa9bbe8a5).